src/request/sign-transaction/SignTransactionApi.js

```diff
@@ -4,26 +4,60 @@
 /* global Nimiq */
 /* global AmountTooSmallError, NetworkMissmatchError */
 /* global SignTransactionWithPassphrase, SignTransactionWithPin */
+/* global TransactionType */
+/* global AddressUtils */
 
 class SignTransactionApi extends PopupApi { // eslint-disable-line no-unused-vars
     /**
      * @param {TransactionRequest} txRequest
      */
     async onRequest(txRequest) {
+        // Validation
+        if (!txRequest) throw new Error('Invalid request');
+
+        if (txRequest.type === TransactionType.EXTENDED) {
+            throw new Error('Extended transaction signing is not yet implemented');
+        } else if (txRequest.type === TransactionType.BASIC) {
+            if (txRequest.sender !== txRequest.signer) {
```

Comment:
might want to normalize

Reply:
I moved the normalization to before these checks.